### PR TITLE
Use Discord Info

### LINKED ISSUE #57: Add profile information from Discord
This is something I've already been working on but now have to kinda redo.

 I plan grabbing this data on the backend/next side during build rather than expecting that the image links will always stay up to date.

Info i plan on storing with users:
- Discord ID

Info I will grab using the discord API
- Avatar
- Banner 
- Nickname/discriminator

### PR DESCRIPTION
**Pull request by joeyballentine**
Note: this only uses discord information that has been manually linked with a user

Uses the Discord API to get the avatar for model authors.

The information is grabbed on the server, but the images aren't actually downloaded, just the URLs. This does leave the possibility of someone changing their avatar and making the URL invalid, though I don't think that's a big enough issue to warrant actually having it statically download all the avatars. But maybe it would be worth it to do that anyway, idk.

image

Resolves #57

### CODE REVIEW

File: src/lib/server/file-data.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import { JsonFile, fileExists } from './fs-util';
 
 const DATA_DIR = './data/';
 const MODEL_DIR = join(DATA_DIR, 'models');
-const USERS_JSON = join(DATA_DIR, 'users.json');
+const USERS_JSON = join(DATA_DIR, 'users-discord.json');
```
Comment (RunDevelopment) [inline src/lib/server/file-data.ts:12]: Why?

File: src/lib/server/discord-data.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import { REST } from '@discordjs/rest';
+import { Routes } from 'discord-api-types/v10';
+import { DiscordData } from '../schema';
+
+const discordApiToken = process.env.DISCORD_API_TOKEN || '';
```
Comment (RunDevelopment) [inline src/lib/server/discord-data.ts:5]: What is that?

File: src/lib/schema.ts [hunk 3]
```diff
@@ -56,8 +56,20 @@ interface OnnxFile extends SingleFile {
     type: 'onnx';
 }
 
+export interface DiscordData {
```
Comment (RunDevelopment) [inline src/lib/schema.ts:59]: This interface isn't used in any `data/*` file, so it shouldn't be here in `schema.ts`.

File: src/lib/server/discord-data.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+const rest = new REST().setToken(discordApiToken);
+
+export interface IDiscordUser {
```
Comment (RunDevelopment) [inline src/lib/server/discord-data.ts:9]: This interface is only used internally.
```suggestion
interface IDiscordUser {
```

**Review by RunDevelopment — CHANGES_REQUESTED** [review]

File: src/lib/server/file-data.ts (hunk 1 above)
Comment (joeyballentine) [reply src/lib/server/file-data.ts:12]: Because I forgot to replace the original once I was finished 😅

File: src/lib/server/discord-data.ts (hunk 2 above)
Comment (joeyballentine) [reply src/lib/server/discord-data.ts:5]: You need a token to use the discord API

Comment (RunDevelopment) [reply src/lib/server/discord-data.ts:5]: So how do you intend to get that token?

Comment (joeyballentine) [reply src/lib/server/discord-data.ts:5]: Oh whoops, I completely forgot to add it to the env in the workflow

**Comment by joeyballentine** [comment]
I'll address the actual linking in a separate PR where I'll make users editable

**Comment by joeyballentine** [comment]
Let's just not do this

**Pull request closed without merging**